### PR TITLE
[semantics] Use static color in Flutter test

### PR DESCRIPTION
**Pull request by CaseyHillers**
Google tracking bug: b/273546981

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by CaseyHillers** [comment]
@dnfield as FYI - I'll need to approve the Flutter golds, which I believe is sufficient test coverage. If you have other ideas around testing this, let me know.

**Review by dnfield — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/semantics_debugger.dart [hunk 1]
```diff
@@ -365,7 +365,10 @@ class _SemanticsDebuggerPainter extends CustomPainter {
     }
     final Rect rect = node.rect;
     if (!rect.isEmpty) {
-      final Color lineColor = Color(0xFF000000 + math.Random(node.id).nextInt(0xFFFFFF));
+      // To minmize changes to goldens tests, use a static color.
```
Comment (christopherfujino) [inline packages/flutter/lib/src/widgets/semantics_debugger.dart:368]:
```suggestion
      // To minimize changes to goldens tests, use a static color.
```

**Comment by pdblasi-google** [comment]
This feels like it incentivizes using golden tests for semantics testing. My understanding is that we wanted to _disincentivize_ that behavior, which the random colors did.

**Comment by Hixie** [comment]
The idea here was to use a predictable color each time (it's fixed if the node ID doesn't change). I'm a bit reluctant for us to drop that entirely; it'll make the display pretty unreadable. Is there some other source we could use that would be less likely to change but would still lead to different colors?

When is the node ID changing in such a way that that is the _only_ change to the output?

File: packages/flutter/lib/src/widgets/semantics_debugger.dart [hunk 2]
```diff
@@ -365,7 +365,10 @@ class _SemanticsDebuggerPainter extends CustomPainter {
...
+      final Color lineColor = const bool.fromEnvironment('FLUTTER_TEST')
```
Comment (christopherfujino) [inline packages/flutter/lib/src/widgets/semantics_debugger.dart:369]: Who provides this dart define?

Comment (christopherfujino) [reply packages/flutter/lib/src/widgets/semantics_debugger.dart:369]: Or was this supposed to be from an environment variable?

Comment (CaseyHillers) [reply packages/flutter/lib/src/widgets/semantics_debugger.dart:369]: Your right, I didn't know this was for dart defines.

Is there an agnostic way to do this in the framework? Pulling in dart:io here would be a blocker

Comment (christopherfujino) [reply packages/flutter/lib/src/widgets/semantics_debugger.dart:369]: Good point, we probably can't make this change in the framework...

File: packages/flutter/lib/src/widgets/semantics_debugger.dart [hunk 3]
```diff
@@ -365,7 +365,12 @@ class _SemanticsDebuggerPainter extends CustomPainter {
     }
     final Rect rect = node.rect;
     if (!rect.isEmpty) {
-      final Color lineColor = Color(0xFF000000 + math.Random(node.id).nextInt(0xFFFFFF));
+      // Passing `--dart-define=static_semnatics=true` will use a static color
```
Comment (christopherfujino) [inline packages/flutter/lib/src/widgets/semantics_debugger.dart:368]:
```suggestion
      // Passing `--dart-define=static_semantics=true` will use a static color
```

**Review by christopherfujino — APPROVED** [review]
RSLGTM

**Comment by CaseyHillers** [comment]
I'm going to close this for now, and may revisit. I did a quick data analysis, and it appears this only fixes about 10% of the problem I see downstream. I'm going to first get buy in internally, and I can upstream this if I make progress.

**Pull request closed without merging**